SpatialGDK/Source/SpatialGDK/Private/Tests/AuthorityRecordTest.cpp

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Tests/TestDefinitions.h"
+
+#include "AuthorityRecord.h"
+
+#define AUTHORITYRECORD_TEST(TestName) \
+	GDK_TEST(Core, AuthorityRecord, TestName)
+
+using namespace SpatialGDK;
+
+namespace
+{
+	class AuthorityChangeRecordFixture
+	{
+	public:
+		const Worker_EntityId kTestEntityId = 1337;
+		const Worker_ComponentId kTestComponentId = 1338;
+
+		const EntityComponentId kEntityComponentId{ kTestEntityId, kTestComponentId };
+
+		AuthorityRecord Record;
+
+		TArray<EntityComponentId> ExpectedAuthorityGained;
+		TArray<EntityComponentId> ExpectedAuthorityLost;
+		TArray<EntityComponentId> ExpectedAuthorityLostTemporarily;
+	};
+} // anonymous namespace
+
+AUTHORITYRECORD_TEST(GIVEN_EmptyAuthorityRecord_WHEN_set_to_authoritative_THEN_AuthorityRecord_has_AuthorityGainedRecord)
```

Comment:
These names are hard to read for a few reasons I think.
There is inconsistency between when pascal case and snake case are used.
They use terminology that isn't present anywhere else.
They don't quite accurately describe things.

Reply:
Things like Empty looks like a type, and authoritative is lower case but then referred to later as AuthorityGainedRecord.

Reply:
It might be easier to say give [state in state machine] when [authority input] then [state in state machine]

Reply:
Ok rereading actually I think they make sense and are sort of what I said. It might just be that I really don't like given when then as a thing to force. Although "AuthorityRecord_has_AuthorityGainedRecord" still takes me a few attempts to parse.

Reply:
I assume this can be resolved.